File: past201912/past201912_b.cs
// https://atcoder.jp/contests/past201912-open/submissions/9525206

using System;

namespace CSharp
{
    class Program
    {
        static void Main()
        {
            var N = int.Parse(Console.ReadLine());

            var A = new int[N];
            for (var i = 0; i < N; i++)
            {
                A[i] = int.Parse(Console.ReadLine());
            }

            for (var i = 0; i < N - 1; i++)
            {
                var diff = A[i + 1] - A[i];
                if (diff > 0)
                {
                    Console.WriteLine($"up {diff}");
                }
                else if (diff < 0)
                {
                    Console.WriteLine($"down {-diff}");
                }
                else
                {
                    Console.WriteLine("stay");
                }
            }
        }
    }
}

File: past201912/past201912_b.fs
// https://atcoder.jp/contests/past201912-open/submissions/13492369

let N = stdin.ReadLine() |> int
let A = Array.zeroCreate<int> N

for i in [ 0..N - 1 ] do
    A.[i] <- (stdin.ReadLine() |> int)

A
|> Array.pairwise
|> Array.map (fun x ->
    match (snd x - fst x) with
    | y when y > 0 -> "up " + y.ToString()
    | y when y < 0 -> "down " + (abs y).ToString()
    | _ -> "stay")
|> Array.iter stdout.WriteLine
